src/core/annotation.js

```diff
@@ -1367,6 +1375,71 @@ class WidgetAnnotation extends Annotation {
     }
     return localResources || Dict.empty;
   }
+
+  _collectJS(entry, xref, list, parents) {
+    if (!entry) {
+      return;
+    }
+
+    let parent = null;
+    if (isRef(entry)) {
+      if (parents.has(entry)) {
+        // If we've already met entry then we've a cycle
+        return;
+      }
+      parent = entry;
+      parents.add(parent);
+      entry = xref.fetch(entry, true);
+    }
+    if (Array.isArray(entry)) {
+      for (const e of entry) {
+        this._collectJS(e, xref, list, parents);
+      }
+    } else if (entry instanceof Dict) {
+      if (entry.get("S") === Name.get("JavaScript") && entry.has("JS")) {
```

Comment:
Sorry, but the `Name` comparison-part needs to read `isName(entry.get("S"), "JavaScript")` instead, since this could potentially break once the `Name`-cache has been reset.